src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/Semantics/ExplicitConversion.cs

```diff
@@ -132,11 +132,14 @@ public bool Bind()
 
                 // But it failed, and so we know that the constant is not in range
 
+                Debug.Assert(
+                    _typeDest.GetTypeKind() != TypeKind.TK_MethodGroupType
+                    && _typeDest.GetTypeKind() != TypeKind.TK_ErrorType
+                    && _typeDest.GetTypeKind() != TypeKind.TK_ArgumentListType
+                    && _typeDest.GetTypeKind() != TypeKind.TK_ParameterModifierType
+                    && _typeDest.GetTypeKind() != TypeKind.TK_NullableType);
```

Comment:
It always strikes me as more of an actual assertion to either say "we never have …" or "we only ever have …".
I more strongly feel so in the case where the whole range of an enum is covered, as "this isn't some very strange out of range value" or where there's a particularly clear reason to rule out a particular value. Here I could go either way, and likely would have done a simpler replace if I hadn't just done some cases that fit the other category, so was keeping with that approach.